### PR TITLE
Feat/check message validity

### PR DESCRIPTION
**Pull request by royscheeren**
# Scope
Make sure the received message is what we expect.

# Work Done
- Replaced the verifySignature function with a verifyLogin function
- The verifyLogin will check for both a valid signature and a valid message format
- using the introduced verifyMessage function

### CODE REVIEW

**Comment by jdsika** [comment]
Should we suggest Marigold as a reviewer here?

File: packages/sdk/src/lib/utils.ts [hunk 1]
```diff
@@ -28,3 +29,21 @@ export const createMessagePayload = (signatureRequestData: SignInMessageData) =>
   )(signatureRequestData)
 
 export const verifySignature = taquitoVerifySignature
+
+export const verifyMessage = (messagePayload: string, pkh: string) => pipe(
```
Comment (jeroenbranje) [inline packages/sdk/src/lib/utils.ts]: We should validate the dappUrl here

**Review by jeroenbranje — DISMISSED** [review]

**Comment by jeroenbranje** [comment]
1 of the test is failing

File: packages/sdk/src/lib/utils.ts [hunk 2]
```diff
@@ -28,3 +29,21 @@ export const createMessagePayload = (signatureRequestData: SignInMessageData) =>
...
+  unpackMessagePayload,
+  ifElse(
+    is(Error),
+    always(false),
+    allPass([
+      propEq('prefix', '0501'),
```
Comment (jdsika) [inline packages/sdk/src/lib/utils.ts]: define a const

**Review by jdsika — DISMISSED** [review]

**Comment by jdsika** [comment]
Can we get a green pipeline? 🥇

**Review by jdsika — APPROVED** [review]

**Comment by tj825** [comment]
LGTM!

**Pull request merged**